src/main/java/fi/aalto/cs/apluscourses/intellij/actions/ResetProjectAction.java

```diff
@@ -0,0 +1,31 @@
+package fi.aalto.cs.apluscourses.intellij.actions;
+
+import static fi.aalto.cs.apluscourses.utils.PluginResourceBundle.getText;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.project.DumbAwareAction;
+import com.intellij.openapi.project.ProjectManager;
+import fi.aalto.cs.apluscourses.intellij.services.PluginSettings;
+import javax.swing.JOptionPane;
+import org.jetbrains.annotations.NotNull;
+
+class ResetProjectAction extends DumbAwareAction {
+
+  @Override
+  public void actionPerformed(@NotNull AnActionEvent e) {
+    var project = e.getProject();
+    if (project == null) {
+      return;
+    }
+
+    if (JOptionPane.showConfirmDialog(null,
+        getText("ui.ResetProjectAction.confirmDialog.content"),
+        getText("ui.ResetProjectAction.confirmDialog.title"),
+        JOptionPane.OK_CANCEL_OPTION) == JOptionPane.OK_OPTION) {
+
+      PluginSettings.getInstance().getCourseFileManager(project).delete();
+
+      ProjectManager.getInstance().reloadProject(project);
```

Comment:
I'm wondering if some concurrent action can possibly re-create the project file between these two statements. But that's probably a rare occurrence without any severe consequences so it doesn't matter.